code/modules/projectiles/gun.dm

```diff
@@ -310,6 +310,8 @@
 	return ..()
 
 /obj/item/gun/proc/try_fire_gun(atom/target, mob/living/user, params)
+	if(SEND_SIGNAL(user, COMSIG_MOB_TRYING_TO_FIRE_GUN, src, target, params) & COMPONENT_CANCEL_FIRING_GUN)
+		return
```

Comment:
I suppose this should be next to `SEND_SIGNAL(src, COMSIG_GUN_TRY_FIRE, ...)` in `fire_gun`?

Reply:
👍